src/components/player-info.js

```diff
@@ -73,8 +75,20 @@ AFRAME.registerComponent("player-info", {
     APP.isAudioPaused.delete(avatarEl);
     deregisterComponentInstance(this, "player-info");
   },
+
+  onAvatarModelLoaded(e) {
+    this.applyProperties(e);
+
+    const modelEl = this.el.querySelector(".model");
+    if (this.isLocalPlayerInfo && e.target === modelEl) {
+      modelEl.object3D.traverse(function(o) {
+        if (o.isSkinnedMesh) o.parent.add(createHeadlessModelForSkinnedMesh(o));
```

Comment:
We allow users to upload non rigged avatars and in that case we won't be hiding the mesh in first person mode.

Reply:
Ah yes good catch. I guess in that case we should still put the whole mesh in the THIRD_PERSON_ONLY layer, but then have no FIRST_PERSON_ONLY mesh.

Reply:
Yep I was thinking on something like that, if there was no headless mesh created just add it to the third person layer only.